test/test_nn.py

```diff
@@ -5801,8 +5801,10 @@ def test_Transformer_cell(self):
         tgt = torch.randn(tgt_length, bsz, d_model)
         tgt_mask = transformer.generate_square_subsequent_mask(tgt_length).double()
         memory_mask = torch.randn(tgt_length, seq_length).double()
+        key_padding_mask = torch.rand(bsz, seq_length) >= 0.5
 
-        output = transformer(src, tgt, src_mask=src_mask, tgt_mask=tgt_mask, memory_mask=memory_mask)
+        output = transformer(src, tgt, src_mask=src_mask, tgt_mask=tgt_mask,
+                             memory_mask=memory_mask, key_padding_mask=key_padding_mask)
         output.sum().backward()
 
     def test_transformerencoderlayer(self):
```

Comment:
Though we didn't do the deterministic test here for the kwarg, it would be great if you could add the deterministic test for key_padding_mask.

Reply:
Added